### ISSUE TITLE
Request for Simpler Dev: Simplify Notification Listeners

### ISSUE THREAD

**Issue by cliftonlabrum** [realm/realm-cocoa#5947]
_(Originally based on [this issue](https://github.com/realm/realm-cocoa/issues/5943).)_

### 🌟Simplify Notification Listeners
Allow me to turn on a feature in a view controller that enables live-listening of all Realm objects and collections. If something changes, auto-refresh the values. Or at least coalesce all object/collection changes into a single closure where I can fire my UI refresh code.

For complex UIs I can have like 9 different listeners running. Setting up all that code with their `NotificationToken`s can be cumbersome. I just wish my collections/objects would just stay up-to-date automagically.

I love Realm. Thank you so much for creating it. Please keep up the great work! 😁👍🏼🙏🏼

**Comment by astigsen** [comment]
Have you thought about just using `Realm.observe()`? It will notify you on any change to the realm, so you could have this call a single closure that updates your UI.

It might give you a few or a lot unneeded updates, depending on your usage patterns, but the way to avoid that is obviously to set up specific listeners for just the changes that could affect that part of the UI as you have already been doing.

**Comment by cliftonlabrum** [comment]
Yeah, I sometimes observe the entire Realm when I can, but I usually have more fine-grained requirements where I need to track a specific collection (and its count) so I can animate changes into a `UITableView` and `NSTableView`.
